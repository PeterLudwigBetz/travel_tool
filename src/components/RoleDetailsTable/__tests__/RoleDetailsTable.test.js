import React from 'react';
import { shallow } from 'enzyme';
import { RoleDetailsTable } from '..';

const props = {
  travelTeamMembers: [{
    fullName: 'A user',
    centers: [
      {
        location: 'Nairobi, Kenya'
      }
    ]
  }],
  roleName: 'Travel team member',
  handleEditCenter: jest.fn(),
  roleUser: {
    id: 1,
    fullName: 'tomato',
    centers: [
      {
        location: 'Nairobi, Kenya'
      }
    ]
  }
};
let wrapper;
describe('<RoleDetailsTable />', () => {
  beforeEach(() => {
    wrapper = shallow (<RoleDetailsTable {...props} />);
  });
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('render the correct number of users for a role', () => {
    expect(wrapper.find('.table__rows').length).toEqual(1);
  });
  it('renders all the columns for a user', () => {
    expect(wrapper.find('.role-user__name')
      .text()).toEqual('A user');
    expect(wrapper.find('.pl-sm-120')
      .text()).toEqual('Nairobi, Kenya');
  });
  it('renders text when there are no users', () => {
    wrapper = shallow (<RoleDetailsTable
      roleUsers={[]}
      roleName="Travel team member" />);
    expect(wrapper.find('.table__requests--empty')
      .text()).toEqual('No travel team member at the moment');
  });
  it('renders error text when there is an error', () => {
    wrapper = shallow (<RoleDetailsTable error="Network error" />);
    expect(wrapper.find('.table__requests--error')
      .text()).toEqual('Network error');
  });

  it('should set `visibility` prop to `visible` when add new role button is clicked', () => {
    const {handleEditCenter } = props;
    wrapper.find('#editButton').simulate('click');
  });
    
});
