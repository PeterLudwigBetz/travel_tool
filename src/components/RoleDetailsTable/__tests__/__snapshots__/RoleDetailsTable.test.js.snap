// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoleDetailsTable /> renders correctly 1`] = `
<Fragment>
  <div
    className="table__container"
  >
    <table
      className="mdl-data-table mdl-js-data-table table__requests"
    >
      <thead>
        <tr>
          <th
            className="mdl-data-table__cell--non-numeric bb-md-0 table__head freeze freeze-head "
          >
            Name
          </th>
          <th
            className="mdl-data-table__cell--non-numeric table__head pl-sm-100d description-left"
          >
            Center
          </th>
          <th
            className="mdl-data-table__cell--non-numeric table__head table__head--last"
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody
        className="table__body"
      >
        <tr
          className="table__row"
        >
          <td
            className="mdl-data-table__cell--non-numeric role-user__name table__data freeze-role-table"
          >
            A user
          </td>
          <td
            className="mdl-data-table__cell--non-numeric table__data pl-sm-120"
          >
            Nairobi, Kenya
          </td>
          <td
            className="mdl-data-table__cell--non-numeric table__requests__status table__data delete"
            style={
              Object {
                "color": "blue",
              }
            }
          >
            Edit              Delete
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</Fragment>
`;
